### ISSUE TITLE
Ionic Package building android with --prod does not change file size or minify javascript

### ISSUE THREAD

**Issue by jgw96** [driftyco/ionic-cli#2009]
_From @kamok on March 16, 2017 16:46_

**Ionic version:**  (check one with "x")
[ ] **1.x**
[x] **2.x**

**I'm submitting a ...**  (check one with "x")
[x] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://forum.ionicframework.com/ or http://ionicworldwide.herokuapp.com/

**Current behavior:**
Two days ago, I built an `.apk` with this exact command, as copied from my ~/.bash_history: `ionic package build android --prod --profile dev_profile --release`. The `.apk` was 6.6 MB big, took 20 seconds, and was booting up in 6 seconds. This is good.

However, today, using the same exact command, using the same exact commit on Git (did a git checkout to the successful build). The `.apk` was 8.8 MB, took 20 seconds, and boot time at 16 seconds. This is bad. 

I tried debugging this to see if somehow my `--prod` flag wasn't working. So I ran another build like so, without the flag: `ionic package build android --profile dev_profile --release`. This yielded a build time of 40 seconds (expected), and the file was, once again 8.8 MB with a slow boot time.

So I can conclude that `--prod` is being noted by the change in build time, but for some reason, AoT compiling and minifaction is not happening. So no change in boot time, no change in file size. 

Only difference I can think of is the `/platform` directory which is gitignored. And the `/node_modules`, which I did a fresh `npm install`. No avail.

Had a co-worker do the same thing with the same repo + same command, adding `--prod` does not seem to change file size at all and other than taking 20 seconds faster to build there's no difference.

**Expected behavior:**
It should work like it did 2 days ago. I used the exact file, exact command. 

**Other information:**

**Ionic info:** (run `ionic info` from a terminal/cmd prompt and paste output below):

```
Your system information:

Cordova CLI: 6.4.0
Ionic Framework Version: 2.0.0
Ionic CLI Version: 2.1.4
Ionic App Lib Version: 2.1.2
Ionic App Scripts Version: 1.1.4
ios-deploy version: 1.9.0
ios-sim version: Not installed
OS: Mac OS X El Capitan
Node Version: v6.9.1
Xcode version: Xcode 8.2.1 Build version 8C1002
```

```
Installed platforms:
  android 6.1.2
  ios 4.3.0
```

_Copied from original issue: driftyco/ionic#10808_

**Comment by jgw96** [comment]
_From @kamok on March 16, 2017 20:3_

In my `config.xml`, I use  `<engine name="android" spec="latest"/>`. I'm guessing this is COULD be the problem. However, having trouble finding out the version history changelog from 3/14 - 3/16 (today). **Edit:**  They did not make any changes. As seen on https://cordova.apache.org/blog/

**Comment by kamok** [comment]
Here's the logs on local. It seems like the issue with `--prod` is not just on Ionic Package. `ionic build android --prod --release" locally also uses dev mode:

```
$ ionic build android --prod --release -- --keystore="election-android.keystore" --storePassword=*****  --password=***** --alias="election-mobile"

> Election-Mobile@1.0.0 ionic:build /Users/kamok/code/election/election-mobile
> ionic-app-scripts build

[17:18:39]  ionic-app-scripts 1.1.4
[17:18:39]  build dev started ...
[17:18:39]  clean started ...
[17:18:39]  clean finished in 4 ms
[17:18:39]  copy started ...
[17:18:39]  transpile started ...
[17:18:43]  transpile finished in 3.67 s
[17:18:43]  preprocess started ...
[17:18:43]  preprocess finished in less than 1 ms
[17:18:43]  webpack started ...
[17:18:43]  copy finished in 3.84 s
[17:18:55]  webpack finished in 11.94 s
[17:18:55]  sass started ...
[17:18:55]  sass finished in 834 ms
[17:18:55]  postprocess started ...
[17:18:55]  postprocess finished in 1 ms
[17:18:55]  lint started ...
[17:18:55]  build dev finished in 16.47 s
[17:18:59]  lint finished in 4.00 s
Running command: /Users/kamok/code/election/election-mobile/hooks/after_prepare/010_add_platform_class.js /Users/kamok/code/election/election-mobile


add to body class: platform-android

Running command: /Users/kamok/code/election/election-mobile/hooks/after_prepare/delete_test_assets.js /Users/kamok/code/election/election-mobile


Test build @ platforms/android/assets/www/build/test does not exist for removal

ANDROID_HOME=/Users/kamok/Library/Android/sdk

JAVA_HOME=/Library/Java/JavaVirtualMachines/jdk1.8.0_111.jdk/Contents/Home

Subproject Path: CordovaLib

Incremental java compilation is an incubating feature.


:preBuild
 UP-TO-DATE

:preReleaseBuild UP-TO-DATE
:checkReleaseManifest

:CordovaLib:preBuild UP-TO-DATE


:CordovaLib:preReleaseBuild UP-TO-DATE
:CordovaLib:checkReleaseManifest


:CordovaLib:prepareReleaseDependencies

:CordovaLib:compileReleaseAidl

UP-TO-DATE
:CordovaLib:compileReleaseNdk UP-TO-DATE

:CordovaLib:compileLint

UP-TO-DATE

:CordovaLib:copyReleaseLint

UP-TO-DATE

:CordovaLib:mergeReleaseShaders

UP-TO-DATE
:CordovaLib:compileReleaseShaders
 UP-TO-DATE
:CordovaLib:generateReleaseAssets
 UP-TO-DATE

:CordovaLib:mergeReleaseAssets

UP-TO-DATE
:CordovaLib:mergeReleaseProguardFiles UP-TO-DATE
:CordovaLib:packageReleaseRenderscript UP-TO-DATE
:CordovaLib:compileReleaseRenderscript

UP-TO-DATE
:CordovaLib:generateReleaseResValues UP-TO-DATE

:CordovaLib:generateReleaseResources UP-TO-DATE
:CordovaLib:packageReleaseResources
 UP-TO-DATE

:CordovaLib:processReleaseManifest

UP-TO-DATE

:CordovaLib:generateReleaseBuildConfig

UP-TO-DATE

:CordovaLib:processReleaseResources

UP-TO-DATE
:CordovaLib:generateReleaseSources UP-TO-DATE

:CordovaLib:incrementalReleaseJavaCompilationSafeguard

UP-TO-DATE

:CordovaLib:compileReleaseJavaWithJavac

UP-TO-DATE

:CordovaLib:processReleaseJavaRes

UP-TO-DATE

:CordovaLib:transformResourcesWithMergeJavaResForRelease

UP-TO-DATE
:CordovaLib:transformClassesAndResourcesWithSyncLibJarsForRelease

UP-TO-DATE
:CordovaLib:mergeReleaseJniLibFolders

UP-TO-DATE

:CordovaLib:transformNative_libsWithMergeJniLibsForRelease

UP-TO-DATE

:CordovaLib:transformNative_libsWithSyncJniLibsForRelease

UP-TO-DATE

:CordovaLib:bundleRelease

UP-TO-DATE

:prepareAndroidCordovaLibUnspecifiedReleaseLibrary

UP-TO-DATE
:prepareReleaseDependencies


:compileReleaseAidl

UP-TO-DATE
:compileReleaseRenderscript

UP-TO-DATE
:generateReleaseBuildConfig

UP-TO-DATE

:generateReleaseResValues

UP-TO-DATE
:generateReleaseResources UP-TO-DATE
:mergeReleaseResources

UP-TO-DATE
:processReleaseManifest

UP-TO-DATE

:processReleaseResources

UP-TO-DATE

:generateReleaseSources UP-TO-DATE

:incrementalReleaseJavaCompilationSafeguard

UP-TO-DATE

:compileReleaseJavaWithJavac

UP-TO-DATE

:compileReleaseNdk
 UP-TO-DATE

:compileReleaseSources UP-TO-DATE


:lintVitalRelease


:mergeReleaseShaders

UP-TO-DATE
:compileReleaseShaders

UP-TO-DATE

:generateReleaseAssets
 UP-TO-DATE

:mergeReleaseAssets

UP-TO-DATE
:transformClassesWithDexForRelease

UP-TO-DATE
:mergeReleaseJniLibFolders

UP-TO-DATE
:transformNative_libsWithMergeJniLibsForRelease

UP-TO-DATE

:processReleaseJavaRes
 UP-TO-DATE
:transformResourcesWithMergeJavaResForRelease
 UP-TO-DATE

:validateSigningRelease
:packageRelease

UP-TO-DATE
:assembleRelease

:cdvBuildRelease

BUILD SUCCESSFUL

Total time: 1.09 secs

Built the following apk(s):
	/Users/kamok/code/election/election-mobile/platforms/android/build/outputs/apk/android-release.apk
	/Users/kamok/code/election/election-mobile/platforms/android/build/outputs/apk/android-release-unsigned.apk
```

**Comment by kamok** [comment]
I figured out the issue. Simply upgrade the `ionic-cli` from `2.1.4` to `2.2.1`. Now build uses production, my `.apk` is minified again, load up time is fast, and I can sleep again. I don't know if updating `npm` had to do with it, but what I did was remove `npm` and reinstall using `homebrew`. Reinstall `cordova` and `ionic`via the new `npm`. Reason I did this was because regular old `npm install -g ionic` didn't work and it wouldn't upgrade away from `2.1.4`.

Strange enough, the building of android ALSO works on ionic package. Before it was also not building with the `--prod` tag, or atleast I'm assuming it because my boot time for my app was slow, and app was not compressed from 8.8MB to 6.6MB. 

Cheers.
